src/protocol/symmetricstate.c

```diff
@@ -281,6 +281,10 @@ int noise_symmetricstate_mix_key
         (state->hash, state->ck, hash_len, input, size,
          state->ck, hash_len, temp_k, key_len);
 
+    /* Truncate temp_k */
+    if (hash_len == 64 && key_len > 32)
```

Comment:
This would seem to prevent the use of symmetric ciphers with more than 256 bits.  While the noise specification may have that restriction, I tried to avoid putting that restriction into the code to support future ciphers with larger keys.  Why is this needed?